packages/react-components/src/components/table-output-component.tsx

```diff
@@ -212,23 +221,64 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
         this.setState({
             tableColumns: this.columnArray
         });
+        if (this.columnApi) {
+            const columns = this.columnApi.getAllColumns();
+            const timestampHeader = columns.find(column => column.getColDef().headerName === 'Timestamp ns');
+            if (timestampHeader) {
+                this.timestampCol = timestampHeader.getColDef().field;
+            }
+        }
+
     }
 
     private async handleTimeSelectionChange(range: TimelineChart.TimeGraphRange) {
         const start = Math.trunc(this.props.range.getstart() + range.start);
-        if (this.lastIndex.timestamp === start) {
+        const end = Math.trunc(this.props.range.getstart() + range.end);
+
+        this.startTimeStamp = Math.min(start, end);
+        this.endTimeStamp = Math.max(start, end);
+        const startIndex = await this.fetchTableIndex(start);
+        const endIndex = await this.fetchTableIndex(end);
+
+        if (this.lastSelectedRange.startTimestamp === start &&
+            this.lastSelectedRange.endTimestamp === end &&
+            startIndex === startIndex && endIndex === endIndex) {
             return;
         }
-        const index = await this.fetchTableIndex(start);
-        if (index && this.gridApi) {
-            this.lastIndex = { timestamp: start, index: index };
+
+        if (this.gridApi && startIndex && endIndex) {
+            this.lastSelectedRange = {
+                startTimestamp: start,
+                endTimestamp: end,
+                startIndex: startIndex,
+                endIndex: endIndex
+            };
             this.gridApi.deselectAll();
-            this.gridApi.ensureIndexVisible(index);
-            const node = this.gridApi.getDisplayedRowAtIndex(index);
-            if (node.id) {
-                node.setSelected(true);
+            this.startVisibleIndex = startIndex;
+            this.selectRows();
+        }
+    }
+
+    private selectRows(): void {
+        if (this.gridApi) {
+
+            if (this.startTimeStamp === this.endTimeStamp && this.startVisibleIndex) {
```

Comment:
Interesting side effect of this code:
If you click between (the timestamps of) events A and B on the time graph, B is selected in the table (next timestamp). But if you select a small time range that is completely between A and B, nothings gets selected.
This may be what we want, I'm not sure.